src/org/parosproxy/paros/core/scanner/VariantAbstractQuery.java

```diff
@@ -123,6 +124,10 @@ protected void setParameters(int type, List<org.zaproxy.zap.model.NameValuePair>
             listParam.add(new NameValuePair(type, nonNullString(parameter.getName()), nonNullString(parameter.getValue()), i));
             i++;
         }
+        if (i == 0) {
+        	// No query params, lets add one just to make sure
+            listParam.add(new NameValuePair(type, "query", "query", i));

```

Comment:
@akmalhisyam Have you tried increasing the time used for such attacks? I think thats the best way to reduce these FPs - https://groups.google.com/d/msg/zaproxy-develop/y-IJDZG5skk/hfS42b_VBAAJ